src/main/java/life/qbic/omero/BasicOMEROClient.java

```diff
@@ -764,6 +769,64 @@ public ByteArrayInputStream getThumbnail(long datasetId, long imageId) {
 
   }
 
+  /**
+   * Associates instrument metadata with an existing image in the OMERO server.
+   * @param instrument containing information to be associated with the image
+   * @param imageId ID of the image that will be annotated
+   */
+  public void associateInstrumentWithImage(Instrument instrument, Long imageId) {
+    if (!isConnected()) {
+      this.connect();
+    }
+    // 1. Check if an image with the id is in OMERO. Throw a runtime exception if not
```

Comment:
is this the best exception to throw? can the image be missing for other reasons?

Reply:
ah i see, there are multiple cases handled